src/app/clusters/tls-certificate-management-server/tls-certificate-management-server.h

```diff
@@ -0,0 +1,346 @@
+    /**
+     * @brief Removes the root certificate with the given ID
+     *
+     * @param[in] matterEndpoint The matter endpoint to query against
+     * @param[in] fabric The fabric the certificate is associated with
+     * @param[in] id The id of the root certificate to remove.
+     * @return CHIP_ERROR_NOT_FOUND if no mapping is found.
+     */
+    virtual CHIP_ERROR RemoveRootCert(EndpointId matterEndpoint, FabricIndex fabric, Tls::TLSCAID id) = 0;
+
+    /**
+     * @brief Generates a client certificate signing request (CSR)
+     *
+     * @param[in] matterEndpoint The matter endpoint to query against
+     * @param[in] fingerprint The fingerprint of the root certificate to find.
+     * @param[out] loadedCallback The lambda to execute with the generated client CSR.
+     * @return NotFound if no mapping is found.
+     */
+    virtual Protocols::InteractionModel::Status GenerateClientCsr(EndpointId matterEndpoint, FabricIndex fabric,
+                                                                  const ClientCsrType & request,
+                                                                  GeneratedCsrCallback loadedCallback) const = 0;
+
+    /**
+     * @brief Appends a client certificate to the provisioned client certificates list maintained by the delegate.
+     *        The delegate must ensure it makes a copy of the provided request and data, if any.
+     *
+     * @param[in] matterEndpoint The matter endpoint to query against
+     * @param[in] fabric The fabric the endpoint is associated with
+     * @param[in] provisionReq The request data specifying the client certificate to be provisioned
+     *
+     * @return CHIP_NO_ERROR if the certificate was appended to the list successfully.
+     * @return CHIP_ERROR if there was an error adding the certificate.
+     */
+    virtual Protocols::InteractionModel::ClusterStatusCode
+    ProvisionClientCert(EndpointId matterEndpoint, FabricIndex fabric, const ProvisionClientCertificateType & provisionReq) = 0;
+
+    /**
+     * @brief Executes loadedCallback for each client certificate.  The certificate passed to loadedCallback has a guaranteed
+     lifetime of the method call.
+     *
+     * @param[in] loadedCallback lambda to execute with allocated memory for client certificate loading.  If this function returns
+     an error result,  returns that same error result.
+
+     * @param[in] matterEndpoint The matter endpoint to query against
+     * @param[in] fabric The fabric the certificate is associated with
+     */
+    virtual CHIP_ERROR LoadedClientCerts(EndpointId matterEndpoint, FabricIndex fabric,
+                                         LoadedClientCertificateCallback loadedCallback) const = 0;
+
+    /**
+     * @brief Executes loadedCallback for all matching client certificates.  The certificates passed to loadedCallback has a
+     guaranteed lifetime of the method call.
```

Comment:
I don't understand this comment, maybe because of number agreement issues?  

The thing passed to loadedCallback is a DataModel::List, right?